.github/workflows/lint.yml

```
@@ -50,46 +50,29 @@ jobs:
         npm_config_yes=true npx github:10up/eslint-json-to-md --path ./eslint-report.json --output ./eslint-report.md
         cat eslint-report.md >> $GITHUB_STEP_SUMMARY
       if: ${{ failure() }}
+
   phpcs:
-    name: phpcs
     runs-on: ubuntu-latest
-
     steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-
-    - name: Set standard 10up cache directories
-      run: |
-        composer config -g cache-dir "${{ env.COMPOSER_CACHE }}"
-
-    - name: Prepare composer cache
-      uses: actions/cache@v2
-      with:
-        path: ${{ env.COMPOSER_CACHE }}
-        key: composer-${{ env.COMPOSER_VERSION }}-${{ hashFiles('**/composer.lock') }}
-        restore-keys: |
-          composer-${{ env.COMPOSER_VERSION }}-
-
-    - name: Set PHP version
-      uses: shivammathur/setup-php@v2
-      with:
-        php-version: '7.4'
-        coverage: none
-        tools: composer:v2, cs2pr
-
-    - name: Install dependencies
-      run: composer install --ignore-platform-reqs
-
-    - name: Check PHPCS standard
-      run: ./vendor/bin/phpcs -i
-
-    - name: Check Compatibility of main plugin file.
-      run: ./vendor/bin/phpcs ./distributor.php --runtime-set testVersion 5.6- --standard=PHPCompatibilityWP
-
-    - name: PHPCS check
-      id: phpcs-sniffs
-      run: ./vendor/bin/phpcs . --report-full --report-checkstyle=./.github/phpcs-report.xml
-
-    - name: Show PHPCS results in PR
-      if: ${{ always() }}
-      run: cs2pr ./.github/phpcs-report.xml
+      - name: Checkout
+        uses: actions/checkout@v2
+      - id: changed-files
+        uses: tj-actions/changed-files@v36
+        with:
+          files: |
+            **/*.php
+      - name: Set PHP version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.4
+          tools: composer:v2, cs2pr
+          coverage: none
+      - name: Install dependencies
+        if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
+        run: composer install
+      - name: Test
+        if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
+        run: ./vendor/bin/phpcs ${{ steps.changed-files.outputs.all_changed_files }} --report-full --report-checkstyle=./.github/phpcs-report.xml --runtime-set testVersion 7.4-
+      - name: Show PHPCS results in PR
+        if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
```

Required to avoid test run stopping in previous step when it fails 

```suggestion
        if: ${{ always() }}
```